pkg/engine/utils/exceptions.go

```diff
@@ -15,8 +15,9 @@ import (
 
 // MatchesException takes a list of exceptions and checks if there is an exception applies to the incoming resource.
 // It returns the matched policy exception.
-func MatchesException(polexs []*kyvernov2.PolicyException, policyContext engineapi.PolicyContext, logger logr.Logger) []kyvernov2.PolicyException {
+func MatchesException(polexs []*kyvernov2.PolicyException, policyContext engineapi.PolicyContext, logger logr.Logger) ([]kyvernov2.PolicyException, []kyvernov2.PolicyException) {
 	var matchedExceptions []kyvernov2.PolicyException
+	var exceptionsForImageVerification []kyvernov2.PolicyException
```

Comment:
There is no need to add the exceptions that set the `verifyImage` to a new list. We can use the existing `matchedExceptions` variable.

And, in the `mutate_image.go`, we can loop through each matched exception and check if it has a `verifyImage`.